homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -35,42 +35,89 @@
     },
 }, extra=vol.ALLOW_EXTRA)
 
+SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_VIN): cv.string,
+})
+
 
 BMW_COMPONENTS = ['binary_sensor', 'device_tracker', 'lock', 'sensor']
 UPDATE_INTERVAL = 5  # in minutes
 
+SERVICE_UPDATE_STATE = 'update_state'
 
-def setup(hass, config):
+_SERVICE_MAP = {
+    'light_flash': 'trigger_remote_light_flash',
+    'sound_horn': 'trigger_remote_horn',
+    'activate_air_conditioning': 'trigger_remote_air_conditioning',
+}
+
+
+def setup(hass, config: dict):
     """Set up the BMW connected drive components."""
     accounts = []
     for name, account_config in config[DOMAIN].items():
-        username = account_config[CONF_USERNAME]
-        password = account_config[CONF_PASSWORD]
-        region = account_config[CONF_REGION]
-        _LOGGER.debug('Adding new account %s', name)
-        bimmer = BMWConnectedDriveAccount(username, password, region, name)
-        accounts.append(bimmer)
-
-        # update every UPDATE_INTERVAL minutes, starting now
-        # this should even out the load on the servers
-
-        now = datetime.datetime.now()
-        track_utc_time_change(
-            hass, bimmer.update,
-            minute=range(now.minute % UPDATE_INTERVAL, 60, UPDATE_INTERVAL),
-            second=now.second)
+        accounts.append(setup_account(account_config, hass, name))
 
     hass.data[DOMAIN] = accounts
 
-    for account in accounts:
-        account.update()
+    def _update_all(_) -> None:
+        """Update all BMW accounts."""
+        for account_update in hass.data[DOMAIN]:
```

Comment:
Below you call the item `bimmer`. Why not call it the same here. It would read nicer.

Reply:
ACK calling the object cd_account (Connected Drive Account).